core/src/test/scala/besom/internal/ReSerializerTest.scala

```diff
@@ -16,14 +17,14 @@ class ReSerializerTest extends munit.FunSuite:
     assertEquals(reSerialize(123.0).unsafeRunSync(), 123.0)
     assertEquals(reSerialize(true).unsafeRunSync(), true)
     assertEquals(reSerialize(false).unsafeRunSync(), false)
-    assertEquals(reSerialize(List.empty[String]).unsafeRunSync(), List.empty[String])
-    assertEquals(reSerialize(List("asdf", "qwer")).unsafeRunSync(), List("asdf", "qwer"))
-    assertEquals(reSerialize(List(1, 2, 3)).unsafeRunSync(), List(1, 2, 3))
-    assertEquals(reSerialize(List(1.0, 2.0, 3.0)).unsafeRunSync(), List(1.0, 2.0, 3.0))
+    assertEquals(reSerialize(Iterable.empty[String]).unsafeRunSync(), List.empty[String])
+    assertEquals(reSerialize(Iterable("asdf", "qwer")).unsafeRunSync(), List("asdf", "qwer"))
+    assertEquals(reSerialize(Iterable(1, 2, 3)).unsafeRunSync(), List(1, 2, 3))
+    assertEquals(reSerialize(Iterable(1.0, 2.0, 3.0)).unsafeRunSync(), List(1.0, 2.0, 3.0))
```

Comment:
these tests are quite funny because they are green only due to iterable apply returning a list from what I recall. I guess you should change the rhs too

Reply:
same here. I also reverted changes in EncoderTest and ResourceDecoderTest